File: src/main/java/com/whitesoft/pinmap/services/UserService.java
package com.whitesoft.pinmap.services;

import com.whitesoft.pinmap.domain.User;

/**
 * Created by borisbondarenko on 18.12.15.
 *
 * Interface for user service (BLL).
 *
 * @author brzzbr
 */
public interface UserService {

    /**
     * get user info by login
     * @param login login
     * @return user object
     */
    User getUser(String login);

    /**
     * get authenticated user
     * @return authenticated user
     */
    User getCurrentUser();
}

File: src/main/webapp/js/services/userService.js
/**
 * Created by borisbondarenko on 25.12.15.
 *
 * @description
 * User service. Allows to get user info.
 *
 * @author brzzbr
 */
(function () {

    angular.module('pinmap')
        .service('userService', ['$resource', 'localStorageService', 'notifyService',userService]);

    function userService($resource, localStorageService, notifyService) {

        var userResource = $resource('/api/user/:id', {id: '@id'});

        return {
            getCurrentUser: getCurrentUser,
            getUser: getUser
        };

        /**
         * Simply gets currnt user
         * @returns {*}
         */
        function getCurrentUser() {

            var currentUserName = localStorageService.get('currentUser');
            return getUser(currentUserName);
        }

        /**
         * Gets some user's info
         * @param username
         * @returns {*}
         */
        function getUser(username) {

            return userResource.get({id: username}).$promise
                .then(function (user) {
                    return user;
                }, function(error){

                    console.log(error);
                    notifyService.error('We have got an error during retrieving user info');
                });
        }
    }
})();
